src/services/mapping/reseller/ExpediaGetMappingService.ts

```diff
@@ -0,0 +1,43 @@
+import { ProductModel, MappingModel } from "@octocloud/generators";
+import { ResellerStatus } from "@octocloud/types";
+import { Reseller } from "../../../types/Reseller";
+import { SpecificResellerGetMappingService } from "./SpecificResellerGetMappingService";
+
+export class ExpediaGetMappingService implements SpecificResellerGetMappingService {
+  public async getMapping(productModels: ProductModel[]): Promise<MappingModel[]> {
+    const mappingModels: MappingModel[] = [];
+
+    productModels.forEach((productModel) => {
```

Comment:
you don't have to use forEach and pushing into a list, you can do it just with `map` and then flatten the array

```ts
    return productModels.map((productModel) => 
      productModel.optionModels.map((optionModel) => 
        optionModel.unitModels.map((unitModel) => {
          const resellerReference = [
            Reseller.Expedia,
            productModel.id,
            optionModel.id,
            unitModel.type.toLowerCase(),
          ].join("_");

          return new MappingModel({
            resellerReference: resellerReference,
            resellerStatus: ResellerStatus.ACTIVE,
            title: productModel.internalName,
            url: "",
            webhookUrl: null,
            optionRequired: true,
            unitRequired: true,
            productId: productModel.id,
            optionId: optionModel.id,
            unitId: unitModel.id,
            connected: Math.random() < 0.5,
            expediaTourTime: null,
            gygPriceOverApi: Math.random() < 0.5,
          });
        })
      )
    ).flat(2);
```